### ISSUE TITLE
Bypass agent and send traces directly to collector?

### ISSUE THREAD

**Issue by jaegchoi** [jaegertracing/jaeger-client-cpp#165]
## Requirement - what kind of business use case are you trying to solve?

We are trying to use Envoy's Jaeger native tracing configuration (https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/jaeger_native_tracing) which uses this C++ client as a plugin. We have a remote secured Jaeger collector that we want to push traces to, but we don't have an agent. We would like to avoid having to set up a local agent as a sidecar, and be able to push our traces to the collector directly.

## Problem - what in Jaeger blocks you from solving the requirement?

It looks like this client requires you to have an agent that will then send the traces to the collector. Is there a way to just configure a remote https reporter/collector so we can directly send spans to it?

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Some other Jaeger clients allow you to set an env variable JAEGER_ENDPOINT that allows spans to be directly submitted to the collector. https://www.jaegertracing.io/docs/1.13/client-features/

## To summarize

This my current understanding of this client:

- samplingServerURL is the collector endpoint where spans will finally end up
- localAgentHostPort is the agent endpoint where the agent will pick up the spans and send along to the samplingServerURL
We would like a way to bypass the agent and send spans directly to the collector
If my understanding is wrong or if this is already possible please let me know

**Comment by mdouaihy** [comment]
Hi @jaegchoi,

I think this will require to have an Http transport in the library to send the spans to the collector address (eg. "http://" + host + ":" + port + "/api/traces")

Do you think you can provide a PR for that?

**Comment by yurishkuro** [comment]
maybe we need a new release before closing?

**Comment by a-sharma11** [comment]
Is the release coming soon for this feature?

**Comment by a-sharma11** [comment]
Bit mis leading here. From the documentation seems like its possible with the current version?
https://github.com/jaegertracing/jaeger-client-cpp#connecting-directly-to-the-collector

**Comment by DanTulovsky** [comment]
Should this be working now with version 0.6.0?  My config is below, but I don't see any traces showing up...

(I do have traces from other apps, unrelated to this tracer...)

```
    {
      "service_name": "nginx",
      "sampler": {
        "type": "const",
        "param": 100
      },
      "reporter": {
        "logSpans": true,
        "endpoint": "http://linkerd-collector.linkerd:14268/api/traces"
      },
      "headers": {
        "jaegerDebugHeader": "jaeger-debug-id",
        "jaegerBaggageHeader": "jaeger-baggage",
        "traceBaggageHeaderPrefix": "uberctx-"
      },
      "baggage_restrictions": {
        "denyBaggageOnInitializationFailure": false,
        "hostPort": ""
      },
    }
```

**Comment by DanTulovsky** [comment]
nginx conf has:

```
   ...
    load_module modules/ngx_http_opentracing_module.so;
   ...
    http {
        # tracing
        opentracing_load_tracer /usr/local/lib/libjaegertracing_plugin.so /etc/jaeger-nginx-config.json;
        opentracing on;
        opentracing_tag http_user_agent $http_user_agent;
       ...
    }
```
